build failed on compass in Win7

mattbrictson opened the pull request:
When I ran mvn jetty:run, I would get a compile error with compass.  It said that it couldn't run compass in a certain directory.  I went to that directory and could run compass just fine.  Compass was on my path.  I noticed that there was a compass and compass.bat in the ruby/bin folder.  So in the pom.xml I changed compass to compass.bat and fixed the problem.

So either the pom.xml needs to know whether to run compass or compass.bat, or I needed to do something differently with my compass install.

mattbrictson commented:
@augustinku we an all-Mac/Linux shop here, so I'm afraid I don't have any experience executing Compass on Windows. It sounds like you are onto a solution, though. Is there any way you could generalize your workaround and contribute it to the fiftyfive-wicket project as a pull request? I would really appreciate the help.

augustinku commented:
I'd be happy to test a solution for win7, but I don't think that I can come up with a fix, since I'm totally unfamiliar with Maven.  This is the first maven project that I've used.  I just changed the two instances of "compass" in the pom.xml to "compass.bat".

mattbrictson committed:
Upgrade ant plugin to version 1.7

mattbrictson committed:
If on Windows, exec compass.bat instead of compass

mattbrictson committed:
Properly escape $ signs

mattbrictson commented:
@augustinku I've attached a potential solution to this issue. Please try editing your POM with the changes shown in commit e319ac8 and see if that works for you. Essentially I am detecting if the environment is Windows and running `compass.bat` instead of `compass`.

mattbrictson commented:
You may also need to upgrade the ant plugin to version 1.7, as shown here: e07f32e.

augustinku commented:
I made a few changes to the code in order to get it to work.  It turns out "property" and "condition property" don't mix well.  If we just use condition property everything works.  

```
            <condition property="compass.executable" value="compass">
               <os family="mac"/>
            </condition>
            <condition property="compass.executable" value="compass">
               <os family="unix"/>
            </condition>
            <condition property="compass.executable" value="compass.bat">
               <os family="windows"/>
            </condition>
```

mattbrictson commented:
Since Windows is the only exception, I would rather not have conditions for all the other operating systems if possible. My understanding of Ant is that properties cannot be changed once set, so perhaps the solution is to just flip the statements around.

Can you try this?

``` xml
<condition property="compass.executable" value="compass.bat">
  <os family="windows"/>
</condition>
<property name="compass.executable" value="compass"/>
```

augustinku commented:
Yes that works.

mattbrictson committed:
Reorder <property> - the first one executed wins

In other words on Windows, the `compass.executable` will be set to `compass.bat` first, and the second `<property>` statement will be ignored. On non-Windows platforms, the second `<property>` will be used, setting `compass.executable` to `compass`.

mattbrictson commented:
Thanks for testing. I will merge this into `master`. I'll plan on issuing a 4.0.1 release of fiftyfive-wicket sometime in the next week or so that includes this fix.
